tephra-hbase-compat-0.96/src/test/java/co/cask/tephra/hbase96/coprocessor/CellSkipFilterTest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.tephra.hbase96.coprocessor;
+
+import org.apache.hadoop.hbase.Cell;
+import org.apache.hadoop.hbase.KeyValue;
+import org.apache.hadoop.hbase.filter.Filter;
+import org.apache.hadoop.hbase.filter.FilterBase;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * HBase 0.96 specific test for testing {@link CellSkipFilter}.
+ */
+public class CellSkipFilterTest {
+  protected static final byte[] FAM = new byte[] {'f'};
+  protected static final byte[] COL = new byte[] {'c'};
+
+  @Test
+  public void testSkipFiltering() throws Exception {
+    Filter filter = new CellSkipFilter(new MyFilter(Filter.ReturnCode.INCLUDE));
+    long timestamp = System.currentTimeMillis();
+    assertEquals(Filter.ReturnCode.INCLUDE, filter.filterKeyValue(newKeyValue("row1", "val1", timestamp)));
+    assertEquals(Filter.ReturnCode.INCLUDE, filter.filterKeyValue(newKeyValue("row1", "val1", timestamp - 1)));
+
+    filter = new CellSkipFilter(new MyFilter(Filter.ReturnCode.INCLUDE_AND_NEXT_COL));
+    assertEquals(Filter.ReturnCode.INCLUDE_AND_NEXT_COL, filter.filterKeyValue(newKeyValue("row1", "val1", timestamp)));
+    assertEquals(Filter.ReturnCode.NEXT_COL, filter.filterKeyValue(newKeyValue("row1", "val1", timestamp - 1)));
```

Comment:
It would be a good idea to verify that not passing same row/column will give the right answer for all cases.